tcms/core/ajax.py

```diff
@@ -120,14 +119,14 @@ def env_values(self):
             )
 
         def tags(self):
-            query = strip_parameters(request, self.internal_parameters)
-            tags = TestTag.objects
+            query = strip_parameters(request.GET, self.internal_parameters)
+            tags = TestCaseTag.objects
```

Comment:
What do you mean by child -> parent relationship?

Whatever the direction of relationship, here, it only needs to query matched tags by name from ``test_tags`` table (the ``TestTag`` model). In contrast, using ``TestCaseTag`` to query tags requires ``inner join`` which is not good. I don't think this change is necessary.